lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2913,6 +2915,341 @@ def permutation_matrices(n):
+                              ~lolims & everymask, ~uplims & everymask)
+                for i, coord in enumerate([x, y, z])]
+            (xl, xh), (yl, yh), (zl, zh) = coorderr
+
+            # draws capmarkers - flat caps othogonal to the error bars
+            if nolims.any() and capsize > 0:
+                lo_caps_xyz = _apply_mask([xl, yl, zl], nolims & everymask)
+                hi_caps_xyz = _apply_mask([xh, yh, zh], nolims & everymask)
+
+                # setting '_' for z-caps and '|' for x- and y-caps;
+                # these markers will rotate as the viewing angle changes
+                cap_lo = art3d.Line3D(*lo_caps_xyz, ls='',
+                                      marker=capmarker[i_zdir],
+                                      **eb_cap_style)
+                cap_hi = art3d.Line3D(*hi_caps_xyz, ls='',
+                                      marker=capmarker[i_zdir],
+                                      **eb_cap_style)
+                self.add_line(cap_lo)
+                self.add_line(cap_hi)
+                caplines.append(cap_lo)
+                caplines.append(cap_hi)
+
+            if (lolims | uplims).any():
+                limits = [
+                    _extract_errs(err*dir_vector[i], coord, uplims, lolims)
+                    for i, coord in enumerate([x, y, z])]
+
+                (xlo, xup), (ylo, yup), (zlo, zup) = limits
+                lomask = lolims & everymask
+                upmask = uplims & everymask
+                lolims_xyz = np.array(_apply_mask([xlo, ylo, zlo], upmask))
+                uplims_xyz = np.array(_apply_mask([xup, yup, zup], lomask))
+                lo_xyz = np.array(_apply_mask([x, y, z], upmask))
+                up_xyz = np.array(_apply_mask([x, y, z], lomask))
+                x0, y0, z0 = np.concatenate([lo_xyz, up_xyz], axis=-1)
+                dx, dy, dz = np.concatenate([lolims_xyz - lo_xyz,
+                                             uplims_xyz - up_xyz], axis=-1)
+                self.quiver(x0, y0, z0, dx, dy, dz,
+                            arrow_length_ratio=arrow_length_ratio,
+                            **eb_lines_style)
+
+            errline = art3d.Line3DCollection(np.array(coorderr).T,
+                                             **eb_lines_style)
+            self.add_collection(errline)
+            errlines.append(errline)
+            coorderrs.append(coorderr)
+
+        coorderrs = np.array(coorderrs)
+
+        def _digout_minmax(err_arr, coord_label):
+            return (np.nanmin(err_arr[:, i_xyz[coord_label], :, :]),
+                    np.nanmax(err_arr[:, i_xyz[coord_label], :, :]))
+
+        minx, maxx = _digout_minmax(coorderrs, 'x')
+        miny, maxy = _digout_minmax(coorderrs, 'y')
+        minz, maxz = _digout_minmax(coorderrs, 'z')
+        self.auto_scale_xyz((minx, maxx), (miny, maxy), (minz, maxz), had_data)
+
+        # Adapting errorbar containers for 3d case, assuming z-axis points "up"
```

Comment:
Would it maybe be better to create an `Errorbar3DContainer` instead?

Reply:
I have tried subclassing `ErrorbarContainer` but it resulted in quite a few errors that I failed to fix. I would suggest postponing that, given that no other axes3d methods rely on containers yet.